src/store/calls.js

```diff
@@ -7,7 +7,7 @@ store.onReset(() => { callHistory = [] })
 
 export default {
   log (testDouble, args, context) {
-    store.for(testDouble).calls.push({ args, context })
+    store.for(testDouble).calls.push({ args, context, cloneArgs: _.cloneDeep(args) })
```

Comment:
Thank you for your prompt reply @searls! The current implementation will always call `cloneDeep`, even when `config.cloneArgs` is `false`. True, it will only use the cloned args when that option is set, but the cloning of the arguments happens all the time. This is also highlighted by the original concern you expressed about this change having a performance impact for calling test doubles with large or hard to deep-clone arguments.

Unfortunately the current workaround was to downgrade back to 3.8.1, as there's no way to control the deep cloning.

I am not necessarily proposing an API change, but rather to see if the cloning could only be done when `cloneArgs` is `true`. From the code I referenced here, it seems that the `config` object is not available, so perhaps there's an architectural limitation?

[Here is a very short reproduction](https://jsfiddle.net/ocb1qagy/2/) that evidences how the cloning might impact a test. Obviously it's a ridiculous example, but in our particular case, there's a chain of references that leads to a similar prop definition, where `cloneDeep` then promptly throws an error.

Now, we would never be able to use the `cloneArgs` config option, that is true, but that is okay, we never needed it anyway, and we feel like the corect handling in those situation would be instead to use `td.matchers.argThat` and assert on particular properties we want to stay the same.

Reply:
Ahhhhhhhh, I get it. 

Yeah this is an ancillary benefit of these changes that's not directly related to the functionality. The goal here that we were solving for was error messages and `td.explain` would lie to you if something passed to a test double was mutated during the test before the printout. Cloning the arguments is a way to ensure that the library's messages are accurate, which is indeed very important (as having test output that says stuff like "expected foo({age: 32}) but was actually foo({age: 32})" can do a lot of arm when debugging)

Yeah, I don't have a great solution here. If the issue is the error, then I think we can wrap cloneDeep in a try-catch and fall back on the original args if cloneArgs is some symbol that represents a failed processing.

Reply:
That doesn't sound like a bad idea actually. Although in our case we would probably prefer to maybe have a feature flag to turn off the deep cloning altogether, as it's something that has potential to slow down the app and maybe even cause  huge memory leaks if a reference to the cloned object is retained somewhere, if the cloning is successful.

In our particular case, where we can't control the objects and their references (they're generated by Ember), we prefer the tradeoff of potentially inaccurate messages instead of the performance impact.

Reply:
Yeah, I understand the performance concern but in general this library really tries to reduce or eliminate global configuration settings wherever possible so I'm not comfortable adding a setting that could decrease the veracity of its messages

Reply:
If you're willing to wrap this in error handling and add a test I'd be happy to pick up a PR and take it from there, though

Reply:
Alright, I will add the try/catch, and we'll see how the test suite performs after that. If the performance hit is major, we'll most likely fork the library and add our own config guard.

Reply:
@searls I opened #469, we can continue the discussion there. Thanks!